static/modules/gamemanager.js

```diff
@@ -175,17 +192,29 @@ class GameManager {
 
     // let all entities take their actions
     for (let i = 0; i < this.entities.length; i++) {
+      if (this.entities[i].pausable && this.gamePause) continue;
+
       // exclude inactive entities
       if (this.entities[i].active) {
         this.entities[i].action();
       }
     }
 
+    // let all Gui elements take their actions
+    for (const guiKey of this.guiElements.keys()) {
+      if (this.guiElements.get(guiKey).active) {
+        this.guiElements.get(guiKey).action();
+      }
+    }
+
     // run a physics step on all entities including particles
     /** @type {Entity[][]} */
     const entityLists = [this.entities, this.particles];
     for (let i = 0; i < entityLists.length; i++) {
+      if (entityLists[i].pausable && this.gamePause) continue;
```

Comment:
What do we think about all these `continues`? Couldn't we just use `filter` to create a temporary list of all unpausable entities if the game is paused and iterate through that?

Reply:
That's 2n vs just n, unless you kept track of pausable entities as you create them.

That being said I would put the this.gamePause first to short circuit the condition fractions of a second faster. Gotta get that HIGH fps